osutil/user.go

```diff
@@ -62,6 +62,11 @@ type AddUserOptions struct {
 // allows as valid usernames
 var IsValidUsername = regexp.MustCompile(`^[a-z0-9][-a-z0-9+._]*$`).MatchString
```

Comment:
do we still have use cases for this? do we need to expand its comment?

Reply:
Right now the thinking is that:
a) IsValidUsername is used to validate the name from osutil.AddUser() which is called by devicestate when adding users from the system-user assertion. Making this slightly more restrictive seems to be good to avoid that those names clash with the "system-usernames" from snap.yaml
b) IsValidSystemUsername is used to validate users from the "system-usernames" snap.yaml via "EnsureUserGroup". It needs to be always a superset of IsValudUsername because e.g. snap-seccomp will use it right now to validate users. 

This all points to a problem with terminology though, we have a "system-user" assertion and a "system-usernames" piece in the snap.yaml. I was thinking to rename them to "IsValidAssertionUsername" and "IsValidSystemUsername" maybe? Or "IsValidAddUsername" and "IsValidEnsureUsername" (but the later feels weird).

Reply:
May leave the first alone and rename IsValidSystemUsername ro IsValidSnapSystemUsername. I think the doc comments of AddUser and EnsureUserGroup should also be updated to clarify their validation rules.

Reply:
Thanks for the great suggestion, I did this now.